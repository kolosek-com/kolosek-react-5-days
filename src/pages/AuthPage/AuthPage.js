import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { saveToken } from '../../reducers/AuthReducer/actions'
import { Redirect } from 'react-router-dom';

class AuthPage extends Component {
  constructor(props) {
    super()

    this.state = {
      token: ''
    }
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmit = (e) => {
    e.preventDefault()

    this.props.saveToken(this.state.token)

    this.setState({
      token: ''
    })
  }

  render() {
    if(this.props.isAuthenticated) {
      return <Redirect
        to='/'
      />
    }

    return (
      <div>
        <h1>CircleCI</h1>
        
        <form onSubmit={this.handleSubmit}>
          <input 
            name="token"
            onChange={this.handleChange}
            placeholder='Enter API token...' 
          />
          <input 
            type='submit' 
            placeholder='Submit' 
            value="Save token" />
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
})

const mapDispatchToProps = dispatch => {
  return bindActionCreators({
    saveToken
  }, dispatch);
}


export default connect(mapStateToProps, mapDispatchToProps)(AuthPage)
