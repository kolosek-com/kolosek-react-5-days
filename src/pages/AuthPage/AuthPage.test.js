import React from 'react';
import renderer from 'react-test-renderer';
import { AuthPage } from './AuthPage'

test('AuthPage', () => {
  const component = renderer.create(
    <AuthPage loginUser={() => {}} />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})