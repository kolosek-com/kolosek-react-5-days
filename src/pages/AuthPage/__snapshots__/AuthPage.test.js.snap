// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPage 1`] = `
<div>
  <h1>
    CircleCI
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      name="token"
      onChange={[Function]}
      placeholder="Enter API token..."
    />
    <input
      placeholder="Submit"
      type="submit"
      value="Save token"
    />
  </form>
</div>
`;
